### DISCUSSION TITLE
Investigate Dask Integration

### DISCUSSION THREAD

**Discussion by madhur-tandon** [pyscript/pyscript#1193] [Proposals]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other feature requests and couldn't find a duplicate (including also the ``type-feature`` tag)
- [X] I confirmed that it's not related to another project area (see the above section)

### What is the idea?

The Idea is to run Dask Client within the Python distribution inside the browser. The client should be able to connect with a remote Dask Cluster and send workloads to be computed and also gather back results.

### Why is this needed

To expand the frontier for data workloads from/inside the browser using Python.

### What should happen?

The mechanism how of a connection is made between the Dask Client and the Dask Cluster should probably change to use WebSockets. Custom patches might also be needed to support Dask Client.

**Comment by FabioRosado** [comment]
Here are some thoughts about supporting Dask after our chat in Slack yesterday.

Providing that all packages are in PyPi then versions shouldn't be a problem, but it's more realistic to think that there will be some packages that will have to be ported to work with Pyodide which may break computations unpredictably.  I've seen dask breaking between different versions of dask/distributed itself, pandas, msgspec, cloudpickle,  and other dependencies,.

I wanted to see what would work out of the box, good news! Dask does, bad news, distributed doesnt 😞  which means that we can't really use dask properly if we can't get the client working 😄 

To be honest, I was expecting a bunch of libraries not working, but it seems there are only two that we can't use because micropip can't find pure wheels of:

- psutil
- tornado

Now on the topic of using websockets, @marcosmoyano added ws support for distributed last year. But as I mentioned in slack, the docs don't really mention this so I wonder if this protocol isn't working properly anymore?

**Comment by marcosmoyano** [comment]
@FabioRosado I have no idea. Check the test results on the last distributed build and see if the websockets tests are running and are green. The code is still there 🤷
